### ISSUE TITLE
[CoE Starter Kit] On Premise Data Gateway Usage

### ISSUE THREAD

**Issue by manuelap-msft** [microsoft/coe-starter-kit#1588]
Can you add metrics and utilization on On-Premise data gateway usage?

**Comment by manuelap-msft** [comment]
I believe the only info we have is if an app uses on premise gateways. We can add reporting for that and investigate what else is available.

**Comment by Jenefer-Monroe** [comment]
blocked by lack of product APIs. put in backlog awaiting these

**Comment by petepuu** [comment]
@manuelap-msft, @Jenefer-Monroe This can be done at least using custom connector and Power BI REST API https://learn.microsoft.com/en-us/rest/api/power-bi/gateways/get-gateways

Documentation is for v1.0 but there seems to be v2.0 API available as that is what PPAC is using to get gateways. v1.0 seems to return only standard gateways but v2.0 also personal and VNET gateways. I created simple custom connector just for testing is this working and can share that with you if needed



v2.0 endpoint is _ttps://api.powerbi.com/v2.0/myorg/gatewayclusters

![image](https://user-images.githubusercontent.com/8307644/193268641-35e8c353-df2a-498a-9d20-32d97bc5cdf9.png)

**Comment by Jenefer-Monroe** [comment]
Thanks for the lead on direction here!

**Comment by DamoBird365** [comment]
Hey folks, I have come hunting for a similar answer, are there plans for the CoE to support reporting on Data Gateway usage in Flows/Power Apps?  Thanks

**Comment by Jenefer-Monroe** [comment]
Thanks for expressing your interest! This hasnt popped yet into our priority stack (we are a very small team) but hearing from people that they want this helps!

**Comment by v7herman4** [comment]
The information is available in json returned for for:
Connector: Power Apps for Admin
Action: Get Connections as Admin

![image](https://github.com/microsoft/coe-starter-kit/assets/89024016/e4a3c388-4ba5-432d-bc56-1ccc83bd1624)

**Comment by joshjacquet** [comment]
Adding **+1** to this. We're relatively new to the CoE/Platform and having easier visibility into connection-level usage of the gateways is big. We got used to having relatively simple management of it on the Power BI side, now are playing catchup on the Apps/Automate side.

**Comment by petepuu** [comment]
@jackpoz you should be able to migrate the gateway to new server. Then I _think_ you do not need to change apps and flows using the gateway

https://learn.microsoft.com/en-us/data-integration/gateway/service-gateway-migrate

**Comment by jackpoz** [comment]
@petepuu thanks for the suggestion, but it's not that simple:
- I'm trying to consolidate multiple data gateways in a single one
- The recovery key required to add a new server to the gateway cluster is long gone. Setting up a new key requires to update all flows anyway.

**Comment by mohamrizwa** [comment]
We noticed that this issue has been pending for quite some time. In the absence of further communication, we are proceeding to close this ticket, assuming that your concerns have been addressed or resolved.
If you continue to experience any difficulties related to this matter, please feel free to open a new support ticket. This will help us ensure prompt and effective assistance, rather than reopening the new Issue.
Thank you for your understanding and cooperation.

**Comment by v7herman4** [comment]
Team wondering if this was ever rolled into a version of the Toolkit? Can't seem to gather that from the comment, looks like it was just closed.

**Comment by jackpoz** [comment]
It's marked as done but it was not implemented.

>We noticed that this issue has been pending for quite some time.
Btw this is not an issue, it's a feature request from the maintainer of the project.
